src/components/input/input-field-icon.tsx

```diff
@@ -1,65 +1,57 @@
-import React, { Dispatch, SetStateAction } from "react";
+import React from "react";
 import { useI18nStore } from "../../i18n/i18n-store.ts";
 
 export interface InputFieldIconProps {
   label: string | React.ReactNode;
-  placeholder: string;
+  placeholder: string | undefined;
   onClick: () => void;
   disabled?: boolean;
-  setDisabled: Dispatch<SetStateAction<boolean>>;
   icon: React.ReactNode;
+  name: string;
 }
 
 const InputFieldIcon: React.FC<InputFieldIconProps> = ({
   label,
   onClick,
   disabled,
-  setDisabled,
+  name,
   icon,
   placeholder,
 }) => {
   const i18n = useI18nStore().i18n();
 
   return (
     <div className="mt-7 flex flex-col">
-      <form
-        className="flex flex-col justify-between gap-x-8"
-        onSubmit={(e) => {
-          e.preventDefault();
-          setDisabled(!disabled);
-        }}
-      >
-        <label className="mb-2 font-semibold">{label}</label>
-        {disabled && (
-          <div className="flex flex-row justify-between gap-x-8">
-            <p className="italic">{placeholder}</p>
-            <button
-              className="self-end text-gdk-blue enabled:hover:text-gdk-light-blue"
-              onClick={onClick}
-            >
-              {icon}
-            </button>
-          </div>
-        )}
-        {!disabled && (
-          <div className="flex flex-col justify-between gap-x-8 md:flex-row">
-            <input
-              className={`h-14 flex-auto rounded-2xl px-3 py-2 outline-1 focus:outline-2
+      <label className="mb-2 font-semibold">{label}</label>
+      {disabled && (
+        <div className="flex flex-row justify-between gap-x-8">
+          <p className="italic">{placeholder}</p>
+          <button
+            className="self-end text-gdk-blue enabled:hover:text-gdk-light-blue"
+            onClick={onClick}
+          >
+            {icon}
+          </button>
+        </div>
+      )}
+      {!disabled && (
```

Comment:
If not disabled? (just a style question but parsing the logic makes this harder IMO)